### DISCUSSION TITLE
import { PrismaClient } from "@prisma/client"

### DISCUSSION THREAD

**Discussion by sdeagh** [prisma/prisma#2222] [Q&A] [answered]
Following the Quickstart guide - Start from scratch (Prisma Generate) the example code for index.js has:

`const { PrismaClient } = require("@prisma/client")`

and this all works fine.  But running `yarn prisma generate` states you can use ES6:

`import { PrismaClient } from '@prisma/client'`

even after adding "type":"module" to my package.json i cant get the import statement to work:

SyntaxError: The requested module '@prisma/client' does not provide an export named 'PrismaClient'

Is it possible to use the ES6 syntax with prisma?

**Comment by Rizialdi** [comment, marked as answer]
Hi,
This error is not related to Prisma. As you can read [here](https://stackoverflow.com/questions/39436322/node-js-syntaxerror-unexpected-token-import) , *import* statement is *not yet supported in NodeJS by default*. The generated message from Prisma in the console is, I think a generic in case you use Typescript an compile it beforehand.:
```js
import { PrismaClient } from '@prisma/client'
// or const { PrismaClient } = require('@prisma/client')
```
For information try to read the starter project using TypeSript and you'll see a *ts.config* file which possesses the instructions for compiling the *index.ts* which contains:
```js
import { PrismaClient } from '@prisma/client'
```

If you really wat to use import/export statements back to Node6+, checkout [esm](https://www.npmjs.com/package/esm) module.

Hope this can help you.

**Reply by Jackbaude** [reply]
Thank you so much. I spent the last hour looking for this fix and this was it. I am so happy right now it's hard to put into words

**Comment by mapoart** [comment]
Hi folks!

It's possible to use easy prisma client with {"type":"module"} without TypeScript. You can do this (you can use both require and import. I am using require only for prisma client in this case):

```js
import { createRequire } from "module";
const require = createRequire(import.meta.url);
const { PrismaClient } = require("@prisma/client");

const prisma = new PrismaClient();
```
Hope this will help you.

**Comment by kacpherek** [comment]
Changing
```
    "target": "es5",
    "module": "esnext",
```
To 
```
    "target": "esnext",
    "module": "commonjs",
```
In tsconfig.json works like a charm for me.
You should give it a shot.

**Reply by brianshimkus** [reply]
Worked like a charm!

**Reply by zhaoxinkun** [reply]
![image](https://github.com/user-attachments/assets/41005031-5cd4-4848-8791-438730b38f80)
It's of no use to me. This problem is giving me a headache.

But I found a new method

You need to ensure that it has been executed.  npm i @prisma/client  npm i  prisma,
and then 

You don't need to update tsconfig.ts
![image](https://github.com/user-attachments/assets/805e7a94-9b3a-44bd-8c12-5f2416fa39e2)

All you need to do is comment out the "output" directory.
![image](https://github.com/user-attachments/assets/e5155bc8-7d73-437d-9d37-caf44eb23990)

and then executed : npx prisma generate  

Everything will be fine now.

**Reply by KaranCodeValley** [reply]
Thanks bro!, It worked for me, thankyou so much I was struggling with this issue for 2 days finally it got solved because of you

**Reply by devluiis** [reply]
aqui deu certo, obrigado!

**Comment by alvescleiton** [comment]
Just changing the attribute "module" in tsconfig.json to "esnext" worked to me!

**Comment by mhiguera** [comment]
I just realized that, even if you have everything configured correctly to load ES modules, if you are using pnpm as your package manager and you remove `node_modules` (or `.pnpm` cache, for that matter), you might end up facing this issue.

Calling `npx prisma generate` again resolves the error:

```✔ Generated Prisma Client (v5.9.1) to ./node_modules/.pnpm/@prisma+client@5.9.1_prisma@5.9.1/node_modules/@prisma/client in 73ms```

**Reply by huykon** [reply]
you save my life ;)

**Reply by thiagocrux** [reply]
Thank you!

**Reply by BlackH3art** [reply]
Thanks for that!

**Comment by glaucusec** [comment]
I am using javascript and not ts. for men using require() inside worked.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!

**Comment by GIT-Saikat** [comment]
This thing was fixed for me by changing the node version to the latest  version in root level . Node version which I used is 24.3.0

**Reply by KhantLoonThu** [reply]
How did it fix? I use Prisma 7 version and got this error.
<img width="702" height="135" alt="Screenshot 2026-01-22 153236" src="https://github.com/user-attachments/assets/8e194d9c-ebd1-4beb-aea2-6d3a2e148ead" />
<img width="857" height="189" alt="Screenshot 2026-01-22 153338" src="https://github.com/user-attachments/assets/e8e7d53f-8fdd-4fe7-a8ed-f7a4ac2f68e5" />

And can't fix it. When I use the path that comes from generated: ../generated/prisma/client, it works (but showing Expected 1 arguments, but got 0.)
<img width="779" height="125" alt="Screenshot 2026-01-22 153515" src="https://github.com/user-attachments/assets/327245c6-556b-46f3-9c96-78e9b219c1af" />

**Reply by KhantLoonThu** [reply]
I found out the problem, I need to pass through an argument in the PrismaClient({ adapter }); as you can see in the documentation.
<img width="432" height="164" alt="Screenshot 2026-01-22 191249" src="https://github.com/user-attachments/assets/aa41047e-eef8-4a8f-8c45-8b99ffcdd21c" />
[https://www.prisma.io/docs/guides/nextjs](url)

**Reply by glaucusec** [reply]
Yeah, they require us to pass is like this in the new version

**Reply by solivares-btf** [reply]
but you did not fix the  import from `@prisma/client? `
